ada.html.markdown

```diff
@@ -0,0 +1,367 @@
+
+         --  Ada 2022 introduced @ for LHS, so the above would be written as
+         --  Counter := @ + 1;  --  Try it, -gnat2022.
+      end loop;
+   end;
+
+   declare
+      package Hue_IO is new IO.Enumeration_IO (Hues);
+
+      --  We can have multiple packages on one line, but I tend to use one
+      --  package per line for readability.
+      use IO, Hue_IO;
+   begin
+      Put ("Hues : ");  --  Note, no prefix.
+
+      --  Because we are using the 'Range attribute, the compiler knows it is
+      --  safe and can omit run-time checks here.
+      for Hue in Hues'Range loop
+         Put (Hue);
+
+         --  Types and objects know about their bounds, their First .. Last
+         --  values. These can be specified as range types.
+         if Hue /= Hues'Last then  --  The /= means "not equal to" like the
+                                   --  maths symbol ≠.
+            Put (", ");
+         end if;
+      end loop;
+
+      IO.New_Line;
+   end;
+
+   --  All objects know their bounds, including strings.
+   declare
+      C : Character := Str (50);  --  Warning caused and exception raised at
+                                  --  runtime.
+      --  The exception raised above can only be handled by an outer scope,
+      --  see [wikipbook](https://en.wikibooks.org/wiki/Ada_Programming/Exceptions#Exception_handlers).
+   begin
+      null;  --  We will never get to this point because of the above.
+   end;
+exception
+   when Constraint_Error =>
+      IO.Put_Line ("Caught the exception");
+end LearnAdaInY;
+```
+
+Now, that's a lot of information for a basic intro to Ada, and I've only touched the surface, there's much more to look at in the references section below. I haven't even touched on dynamic memory allocation which includes [pools](https://ada-lang.io/docs/arm/AA-13/AA-13.11), this is because for the most part, Ada programs don't need it, you can do a lot without it.
+
+As I stated above, Ada barely looks like Pascal and if you look at the original [Green specification](https://apps.dtic.mil/sti/trecms/pdf/ADB950587.pdf) (Warning: Huge 4575 page scanned PDF - starting on page 460), it looks nothing like it at all (page 505 of that PDF).
+
+The above source code will compile, but also will give warnings showing the power of the strong static type system.
+
+## Download this source
+
+If you already have the GNAT toolchain installed, you can cut and paste the above into a new file, e.g. ```learn-ada-in-y.ada``` and then run the following:
+
+```bash
+$ gnatchop learn-ada-in-y.ada # This breaks the program into its specificication ".ads" and body ".adb".
+$ gnatmake empty.adb # gnatmake takes care of compilation of all units and linking.
+$ gnatmake hello.adb
+$ gnatmake learnadainy.adb
+```
+
+Or, download [Alire](https://alire.ada.dev), copy it to somewhere in your PATH and then do this:
+
+```bash
+$ alr toolchain --select    # Select the GCC and GPR tools to install
```

Comment:
This not necessary as Alire 1 will ask the user anyway, and Alire 2 will use defaults.